include/ginkgo/core/log/logger.hpp

```diff
@@ -422,6 +423,28 @@ public:                                                              \
                                            one_changed, all_converged);
     }
 
+    /**
+     * Register the `iteration_complete` event which logs every completed
+     * iterations.
+     *
+     * @param solver  the solver executing the iteration
+     * @param b  the right-hand-side vector
+     * @param x  the solution vector
+     * @param it  the current iteration count
+     * @param r  the residual (optional)
+     * @param tau  the residual norm (optional)
```

Comment:
nit: maybe like above (criterion)
```suggestion
     * @param tau  the implicit residual norm squared (optional)
```